src/inputs/RichTextField.tsx

```diff
@@ -63,28 +63,35 @@ export function RichTextField(props: RichTextFieldProps) {
   const onFocusRef = useRef<RichTextFieldProps["onFocus"]>(onFocus);
   onFocusRef.current = onFocus;
 
-  useEffect(
-    () => {
-      if (readOnly) return;
+  // Generate a unique id to be used for matching `trix-initialize` event for this instance.
+  const id = useMemo(() => {
+    if (readOnly) return;
 
-      const editorElement = document.getElementById(`editor-${id}`);
-      if (!editorElement) {
-        throw new Error("editorElement not found");
-      }
+    const id = `${new Date().getTime()}-${Math.floor(Math.random() * 1000)}`;
 
-      editor.current = (editorElement as any).editor;
-      if (!editor.current) {
-        throw new Error("editor not found");
-      }
-      if (mergeTags !== undefined) {
-        attachTributeJs(mergeTags, editorElement!);
+    function onEditorInit(e: Event) {
+      const elId = (e.target as HTMLElement).id;
+      if (elId === `trix-editor-${id}`) {
+        editor.current = (editorElement.current as any).editor;
```

Comment:
I'm kind surprised this works... Like I thought the "attaching `trix-initialize` to `window` in an useMemo" was we needed to attach the event listener _immediately_ because anything in `useEffect` (or anything that uses a ref) would miss the `trix-initialize` event? 

But if `editorElement.current` is correctly assigned here, that means it's ref was assigned before this was triggered...could we just attach `trix-initialize` directly to `editorElement.current` in a `useEffect`? 

Also, if that doesn't work, we could probably use `e.target === editorElement.current` as the "is this the `trix-initialize` event for my element" instead of the random id?...

Reply:
Good question, and I wonder if the `strictMode` flag was messing up my testing for this...
I had put a log statement in the trix code when the `trix-initialize` is called, and that happens before the `useEffect` is executed.
But good question on "how is the ref available then?". Seems like this is a potential race condition...

Reply:
Maybe just `editorElement.current` directly from `e.target`, and skip the ref all together?

Reply:
Just to make sure I thinking through it correctly - If I set the `editorElement.current` directly form `e.target`, then I'll want to keep that random id around as well to ensure I am setting the ref to the proper element.

Reply:
@bdow yeah, that sounds right

Reply:
Also, the "random" id really doesn't have to be random per se, just unique, i.e. you could do:

```
let trixId = 1;
```

As a top-level module const and just use `trixId++` for each new trix instance.